### DISCUSSION TITLE
Token Exchange for v2 to v3 is not working

### DISCUSSION THREAD

**Discussion by danscan1712** [etsy/open-api#944] [Q&A]
https://api.etsy.com/v3/public/oauth/token?grant_type=token_exchange&client_id=XXXXXXXXXc4xkt82lj4&legacy_token=XXXXXXXXXXXf1efac

I am making a post request from postman to get the new keys but I get the following response:
{
    "error": "invalid_request",
    "error_description": "grant_type is required"
}

the client_id is my App Id and the legacy token is the Token I would use in V2.

**Comment by ablagoev** [comment]
Hey, are you sure you are doing a POST request and are passing the parameters in the post body as application/x-www-form-urlencoded?

Not sure if you just pasted the URL above, but it will not work if you pass the parameters in the query string.

On my side the exchange token method works properly and have managed to exchange the tokens of 3 different apps.

Hope this helps.

Cheers,
Alex
